server/util/mdutil/mdutil.go

```diff
@@ -0,0 +1,22 @@
+// mdutil contains utilities for working with gRPC metadata.
+package mdutil
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/metadata"
+)
+
+// Parse parses gRPC metadata from a list of strings in KEY=VALUE format.
+func Parse(kvs ...string) (metadata.MD, error) {
```

Comment:
wonder if this would be better in server/util/bazel_request/bazel_request.go

Reply:
wdyt about merging `bazel_request.go` into this package, since it's not bazel-specific at this point? (I have an itch to rename that package every time I look at it :sweat_smile:)

Reply:
that sounds fine with me if you want to do it in a separate CL